### ISSUE TITLE
Flase positive when importing modules/functions in functions (undefined-value)

### ISSUE THREAD

**Issue by cl-ement05** [PyCQA/pylint#4351]
### Steps to reproduce

Given a file `test.py`:
```python
def import_func():
    global math
    try:
        import math
    except ImportError: 
        print("There was a fatal import error")
        exit()
if __name__ == '__main__':
    import_func()
    math.sqrt(4)

```

### Current behavior
At line 8 : pylint underlines in red saying that math is undefined variable :
![image](https://user-images.githubusercontent.com/64481347/114683612-7604dd80-9d10-11eb-80ef-870ce6bb4f42.png)

Result of `pylint test.py`:
```
************* Module test
test.py:1:0: C0111: Missing module docstring (missing-docstring)
test.py:1:0: C0111: Missing function docstring (missing-docstring)
test.py:2:4: C0103: Constant name "math" doesn't conform to UPPER_CASE naming style (invalid-name)
test.py:2:4: W0603: Using the global statement (global-statement)
test.py:10:4: E0602: Undefined variable 'math' (undefined-variable)

-------------------------------------------------------------------
Your code has been rated at 1.00/10 (previous run: -4.00/10, +5.00)
```

### Expected behavior
Pylint not saying math is undefined
### pylint --version output

Result of `pylint --version` output:

```
pylint3 2.2.2
astroid 2.1.0
Python 3.7.3 (default, Jan 22 2021, 20:04:44)
```

**Comment by Pierre-Sassoulas** [comment]
pylint would require control flow to understand this code properly and we already have ticket for that, so I'm closing.

As an aside, the code is convoluted and hard to understand by human and machine alike. Remove the global and do not catch the error to replace it by an inferior one and you get:

```python
import math

if __name__ == '__main__':
    math.sqrt(4)
```

Which will be superior because you will have a proper stack-trace if math is not importable instead of a vague print and an exit code 0.

**Comment by cl-ement05** [comment]
> Which will be superior because you will have a proper stack-trace if math is not importable instead of a vague print and an exit code 0.
The problem is that I am trying to be as user-friendly as possible on my Python project so dumping the stack trace would be a bad idea. Moreover, the example with the math module was kinda useless because there is like 0.0001% of chance there wil be an ImportError. 

This block of code has much more sense in my project where I use modules that need to be downloaded using pip. SO basically what I do is try to import the module, and if python catches `ImportError`, install the module (using `os.system()`).

**Comment by Pierre-Sassoulas** [comment]
OK I understand better. Did you try defining the ``install_require`` in the packaging of your project ? (https://github.com/PyCQA/pylint/blob/master/setup.cfg#L44).

**Comment by cl-ement05** [comment]
> OK I understand better. Did you try defining the `install_require` in the packaging of your project ? (https://github.com/PyCQA/pylint/blob/master/setup.cfg#L44).

Well no since it is not a pip package, just a simple main.py, 1000 lines long

**Comment by Pierre-Sassoulas** [comment]
Or you can create a `requirements.txt` with your dependencies and `pip3 install requirements.txt` is a pre-requisite of running your script :)

**Comment by cl-ement05** [comment]
> Or you can create a `requirements.txt` with your dependencies and `pip3 install requirements.txt` is a pre-requisite of running your script :) 

Oh thanks I will try that one I didn't know it was possible 🙂
